src/Entity/Transformer.php

```diff
@@ -239,31 +190,20 @@ public function getChildren(PageInterface $page): Collection
     }
 
     /**
-     * @return array<string, mixed>
-     *
-     * @phpstan-return BlockContent
+     * @return \Closure[]
      */
-    private function createBlock(BlockInterface $block): array
+    private function getDenormalizeCallbacks(): array
     {
-        $childBlocks = [];
+        $result = [
+            'position' => static fn (string|int|null $innerObject, string $outerObject, string $attributeName, ?string $format = null, array $context = []): int => null === $innerObject ? 0 : (int) $innerObject,
```

Comment:
@eerison had string for position values for some reason i think

theoretically it should only be int stored as string at max

Reply:
ok

Reply:
found a test case that causes the problem, there is a case for "position" as "" empty string

Reply:
Maybe we should use `int|empty-string|null` then (or `int|''|null`) instead of 
https://github.com/sonata-project/SonataPageBundle/pull/1522/commits/b10100b31f2e8883db0aafb03b945022eeb0ee49
if it's the only accepted value (?)